File: LeetCode/LeetCode 107. 二叉树的层次遍历 II.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution:
    def levelOrderBottom(self, root: TreeNode) -> List[List[int]]:
        ans = list()
        if not root:
            return ans
        
        ss = collections.deque([root])
        while ss:
            n = len(ss)
            now = list()
            for i in range(n):
                k = ss.popleft()
                now.append(k.val)
                if k.left:
                    ss.append(k.left)
                if k.right:
                    ss.append(k.right)
            ans.append(now)
        
        return ans[::-1]

File: LeetCode/LeetCode 107. 二叉树的层次遍历 II.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
 * };
 */
class Solution {
public:
    vector<vector<int>> levelOrderBottom(TreeNode* root) {
        vector <vector<int>> ans;
        queue <TreeNode*> ss;
        if(!root) return ans;
        ss.push(root);
        while(!ss.empty()) {
            int n = ss.size();
            ans.push_back(vector <int> ());
            for(int i = 1; i <= n; i++) {
                auto k = ss.front();
                ss.pop();
                ans.back().push_back(k->val);
                if(k->left) ss.push(k->left);
                if(k->right) ss.push(k->right);
            }
        }
        reverse(ans.begin(), ans.end());
        return ans;
    }
};